thrift/lib/cpp2/test/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This CMakeLists.txt takes care of building a selection of benchmarks and tests in this folder.
+# Currently, it only builds ProtocolBench with its dependencies
+
+# obtain paths to other generated thrift files from the global property space
+get_property(protocol-cpp2-SOURCES GLOBAL PROPERTY
+    protocol-cpp2-SOURCES)
+get_property(protocol_detail-cpp2-SOURCES GLOBAL PROPERTY
+    protocol_detail-cpp2-SOURCES)
+get_property(frozen-cpp2-SOURCES GLOBAL PROPERTY
+    frozen-cpp2-SOURCES)
+get_property(field_mask-cpp2-SOURCES GLOBAL PROPERTY
+   field_mask-cpp2-SOURCES)
```

Comment:
Why get sources instead of linking with relevant targets?

Reply:
I tried to be consistent with the rest of the codebase here. It seems to be implemented this way in the other instances:
https://github.com/facebook/fbthrift/blob/main/thrift/lib/cpp2/CMakeLists.txt#L37
https://github.com/facebook/fbthrift/blob/main/thrift/lib/cpp2/CMakeLists.txt#L101

The `thrift_generate` target also does not add a CMake target as far as I understand